### PR TITLE
Added support for Cygwin.

### PR DESCRIPTION
**Pull request by wvenialbo**
Signed-off-by: Waldemar Villamayor-Venialbo <wvenialbo@gmail.com>

### CODE REVIEW

File: include/termcolor/termcolor.hpp [hunk 1]
```diff
@@ -39,7 +39,13 @@
 
 
 #include <iostream>
+#ifdef __STRICT_ANSI__  // Required when compiling with
```
Comment (ikalnitsky) [inline include/termcolor/termcolor.hpp]: Hm.. Do we really need this in all cases? Maybe just make that workaround only if both `__STRICT_ANSI__` and `__CYGWIN__` are defined?

**Comment by ikalnitsky** [comment]
@wvenialbo thank you for the patch! Could you please answer to my comment inline?

File: include/termcolor/termcolor.hpp (hunk 1 above)
Comment (wvenialbo) [inline include/termcolor/termcolor.hpp]: I did some testing and yes, you are right, it seems that the standard library in Cygwin is stricter than in other unix-like environments. And worst, this workaround won't work when _cstdio_ is included before _termcolor.h_.

A more correct approach when requiring strict standard conformance will be to `#undef __STRICT_ANSI__` whithin the _.cpp_ file that includes _termcolor.h_, above the inclusion of any header that may include _cstdio_, such as _iostream_, or above `#include <cstdio>` itself. That way we ensure the order of the `#include` directives won't have any impact on the compilation unit.

Please dismiss this request, I'll patch and send you a new one.

Results of my tests:

**g++ 4.9, 5.2, 5.3, and clang++ 3.7 on Cygwin 2.4:**
````sh
$ g++ -std=c++11 -c -o test.o test.cpp       : **ERROR**
$ g++ -std=gnu++11 -c -o test.o test.cpp     : **OK**
$ clang++ -std=c++11 -c -o test.o test.cpp   : **ERROR**
$ clang++ -std=gnu++11 -c -o test.o test.cpp : **OK**
````

**g++ 4.8 and clang++ 3.4 on FreeBSD 10.1:**

````sh
$ g++ -std=c++11 -c -o test.o test.cpp       : **OK**
$ g++ -std=gnu++11 -c -o test.o test.cpp     : **OK**
$ clang++ -std=c++11 -c -o test.o test.cpp   : **OK**
$ clang++ -std=gnu++11 -c -o test.o test.cpp : **OK**
````

**g++ 4.8 on Ubuntu 14.04:** (thanks to [jmaidana](https://github.com/jmaidana18))

````sh
$ g++ -std=c++11 -c -o test.o test.cpp       : **OK**
$ g++ -std=gnu++11 -c -o test.o test.cpp     : **OK**
````

In all cases the error was:

````sh
In file included from test.cpp:13:0:
termcolor.hpp: In function 'bool termcolor::__internal::is_atty(const ostream&)':
termcolor.hpp:449:46: error: 'fileno' was not declared in this scope
             return ::isatty(fileno(std_stream));
                                              ^
````

Comment (wvenialbo) [inline include/termcolor/termcolor.hpp]: I forget to mention, BSD tests required a further patch, which was not included here.

**Pull request closed without merging**